analytics-integrations/kahuna/src/main/java/com/segment/analytics/internal/integrations/KahunaEcommerceHelper.java

```diff
@@ -0,0 +1,115 @@
+package com.segment.analytics.internal.integrations;
+
+import android.text.TextUtils;
+
+import com.kahuna.sdk.KahunaAnalytics;
+import com.segment.analytics.internal.model.payloads.TrackPayload;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+class KahunaEcommerceHelper {
+
+  // Segment constants
+  private static final String SEG_VIEWED_PRODUCT_CATEGORY = "Viewed Product Category";
+  private static final String SEG_VIEWED_PRODUCT = "Viewed Product";
+  private static final String SEG_ADDED_PRODUCT = "Added Product";
+  private static final String SEG_COMPLETED_ORDER = "Completed Order";
+  private static final String SEG_DISCOUNT_KEY = "discount";
+
+  // Kahuna constants
+  private static final String CATEGORIES_VIEWED = "Categories Viewed";
+  private static final String LAST_VIEWED_CATEGORY = "Last Viewed Category";
+  private static final String LAST_PRODUCT_VIEWED_NAME = "Last Product Viewed Name";
+  private static final String LAST_PRODUCT_VIEWED_ID = "Last Product Viewed Id";
+  private static final String LAST_PRODUCT_ADDED_TO_CART_NAME = "Last Product Added To Cart Name";
+  private static final String LAST_PRODUCT_ADDED_TO_CART_CATEGORY
+          = "Last Product Added To Cart Category";
+  private static final String LAST_PURCHASE_DISCOUNT = "Last Purchase Discount";
+  private static final int MAX_CATEGORIES_VIEWED_ENTRIES = 50;
```

Comment:
We limit to 50 because we don't wanna let this grow unbounded client side and anything more than 50 starts becoming not as useful from a Kahuna campaign perspective. For this initial phase of the integration, we decided to track 50 unique categories while keeping in mind to not make things too complicated as we are also going to release a newer SDK soon with newer APIs that will make future segment iterations much better. Great question though!